src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -109,22 +112,205 @@ export default class NativeAdapter implements IMediaSourceAdapter {
+          this._eventManager.unlisten(this._videoElement, PlayerEvents['ERROR']);
+          NativeAdapter._logger.error(error);
+          reject(error);
+        });
+        this._videoElement.src = this._source;
+      });
+    }
+    return this._loadPromise;
   }
 
   /**
-   * Clearing the video source
+   * Clear the video source
    * @function destroy
    * @returns {void}
    */
   destroy(): void {
     NativeAdapter._logger.debug('destroy');
-    this._videoElement.src = "";
+    this._eventManager.destroy();
+    this._loadPromise = null;
+  }
+
+  /**
+   * Get the parsed tracks
+   * @function _parsedTracks
+   * @returns {Array<Track>} - The parsed tracks
+   * @private
+   */
+  get _parsedTracks(): Array<Track> {
+    return this._parsedVideoTracks.concat(this._parsedAudioTracks).concat(this._parsedTextTracks);
+  }
+
+  /**
+   * Get the parsed video tracks
+   * @returns {Array<Track>} - The parsed video tracks
+   * @private
+   */
+  get _parsedVideoTracks(): Array<Track> {
+    let videoTracks = this._videoElement.videoTracks;
+    let parsedTracks = [];
+    if (videoTracks) {
+      for (let i = 0; i < videoTracks.length; i++) {
+        let settings = {
+          id: videoTracks[i].id,
+          active: videoTracks[i].selected,
+          label: videoTracks[i].label,
+          language: videoTracks[i].language,
+          index: i
+        };
+        parsedTracks.push(new VideoTrack(settings));
+      }
+    }
+    return parsedTracks;
+  }
+
+  /**
+   * Get the parsed audio tracks
+   * @returns {Array<Track>} - The parsed audio tracks
+   * @private
+   */
+  get _parsedAudioTracks(): Array<Track> {
+    let audioTracks = this._videoElement.audioTracks;
+    let parsedTracks = [];
+    if (audioTracks) {
+      for (let i = 0; i < audioTracks.length; i++) {
+        let settings = {
+          id: audioTracks[i].id,
+          active: audioTracks[i].enabled,
+          label: audioTracks[i].label,
+          language: audioTracks[i].language,
+          index: i
+        };
+        parsedTracks.push(new AudioTrack(settings));
+      }
+    }
+    return parsedTracks;
+  }
+
+  /**
+   * Get the parsed text tracks
+   * @returns {Array<Track>} - The parsed text tracks
+   * @private
+   */
+  get _parsedTextTracks(): Array<Track> {
+    let textTracks = this._videoElement.textTracks;
+    let parsedTracks = [];
+    if (textTracks) {
+      for (let i = 0; i < textTracks.length; i++) {
+        let settings = {
+          kind: textTracks[i].kind,
+          active: textTracks[i].mode === 'showing',
+          label: textTracks[i].label,
+          language: textTracks[i].language,
+          index: i
+        };
+        parsedTracks.push(new TextTrack(settings));
+      }
+    }
+    return parsedTracks;
+  }
+
+  /**
+   * Select a track
+   * @function selectTrack
+   * @param {Track} track - the track to select
+   * @returns {boolean} - success
+   * @public
+   */
+  selectTrack(track: Track): boolean {
```

Comment:
This method should be in the player/engine level since all the adapter has the same logic for it.
The only difference between the adapters is the implementation of the specific selection - i.e `selectAudioTrack, selectVideoTrack, selectTextTrack`and therefore this should be the api that the adapter will implement and the engine will call to.